cachi2/core/package_managers/pip.py

```diff
@@ -58,6 +61,35 @@
 )
 
 
+def fetch_pip_source(request: Request) -> RequestOutput:
```

Comment:
We should probably write a unit test to cover this function.

Reply:
You were a few minutes faster, done :P

The PR is still a draft, but I think the unit test was one of the last things missing. Other than the `dev` key missing in output.json, which I discovered while writing the unit test.